kong/plugins/acl/handler.lua

```diff
@@ -30,9 +30,15 @@ function ACLHandler:access(conf)
   ACLHandler.super.access(self)
 
   local consumer_id
-  if ngx.ctx.authenticated_credential then
+  if ngx.ctx.authenticated_consumer then
+    consumer_id = ngx.ctx.authenticated_consumer.id
+  end
+
+  if not consumer_id and ngx.ctx.authenticated_credential then
     consumer_id = ngx.ctx.authenticated_credential.consumer_id
-  else
+  end
+
```

Comment:
Wouldn't it be simpler to write an `elseif` here?

```
if ngx.ctx.authenticated_consumer then
  consumer_id = ngx.ctx.authenticated_consumer.id
elseif ngx.ctx.authenticated_credential then
  consumer_id = ngx.ctx.authenticated_credential.consumer_id
end
```

Or even

```
local consumer_id = (ngx.ctx.authenticated_consumer and ngx.ctx.authenticated_consumer.id) or
                    (ngx.ctx.authenticated_credential and ngx.ctx.authenticated_credential.consumer_id)
```

(That second one might be a bit too much)